service/query/ons.go

```diff
@@ -55,6 +55,35 @@ func (sv *Service) ONS_GetDomainByOwner(req client.ONSGetDomainsRequest, reply *
 	return nil
 }
 
+func (sv *Service) ONS_GetParentDomainByOwner(req client.ONSGetDomainsRequest, reply *client.ONSGetDomainsReply) error {
+	domains := sv.ons
+	if req.Owner == nil {
+		return codes.ErrBadOwner
+	}
+	ds := make([]ons.Domain, 0)
+
+	domains.Iterate(func(name ons.Name, domain *ons.Domain) bool {
+
+		if domain.Owner.Equal(req.Owner) {
+			if req.OnSale && !domain.OnSaleFlag {
+				return false
+			}
+			if domain.Name.IsSub() {
+				return false
+			}
+			ds = append(ds, *domain)
+		}
+		return false
+	})
+
+	*reply = client.ONSGetDomainsReply{
+		Domains: ds,
+		Height:  sv.balances.State.Version(),
```

Comment:
I think you meant to use:
sv.domains.State.Version()

Reply:
yeah, I'll also change other domain query, since they share same problems.

Reply:
actually it's sv.ons.State.Version()